python/sdk/merlin/requirements.py

```diff
@@ -0,0 +1,285 @@
+import os
+from collections import namedtuple
+from itertools import filterfalse
+from pathlib import Path
+from typing import Any, List, Optional
+
+import yaml
+from packaging.requirements import Requirement
+
+_CONSTRAINTS_FILE_NAME = "constraints.txt"
+
+
+def process_conda_env(
+    conda_env: Any,
+    python_version: Optional[str] = "3.10.*",
+    additional_merlin_reqs: Optional[List[str]] = None,
+):
+    """
+    Processes `conda_env` passed to `merlin.log_pyfunc_model` and `merlin.log_model`.
+    This function will do the following:
+    1. Add/replace Python version on the conda_env with `python_version` argument.
+    2. Add `additional_merlin_reqs` onto the conda_env's pip dependencies.
+
+    :param conda_env: Either a dictionary representation of a conda environment or the path to a conda environment yaml file
+    :param python_version: The python version to replace the conda environment with
+    :param additonal_merlin_reqs: A list of additional Merlin requirements to be added to the conda environment
+    :return: dict representation of the conda environment
+    """
+
+    if isinstance(conda_env, str):
+        with open(conda_env, "r") as f:
+            conda_env = yaml.safe_load(f)
+    elif not isinstance(conda_env, dict):
+        raise TypeError(
+            "Expected a string path to a conda env yaml file or a `dict` representing a conda env, "
+            "but got `{}`".format(type(conda_env).__name__)
+        )
+
+    if conda_env is None:
+        conda_env = {}
+
+    if "dependencies" not in conda_env:
+        conda_env["dependencies"] = []
+
+    conda_env = _overwrite_python_version(conda_env, python_version)
```

Comment:
this will use the python version of the environment that the sdk is being called? meaing to say, if users specific python=3.8 in their conda, it will be ignored

Reply:
correct. this is the current behavior that merlin has. if user specified python=3.8 in their conda but the merlin-sdk is running in python=3.10, merlin will overwrite it and use python=3.10